### ISSUE TITLE
IPC Connection Issue with Waybar and Zen Browser (AUR Version)

### ISSUE THREAD

**Issue by 0x4a61636f62** [zen-browser/desktop#1992]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

**Description:**
I am encountering an IPC connection issue between Waybar and Zen Browser when using the AUR package for Zen Browser on Arch Linux (I am not using the Flatpak version). Specifically, when the Zen Browser is running, restarting Waybar results in the following trace:

```
[2024-10-09 23:03:35.008] [info] Hyprland IPC starting
[2024-10-09 23:03:35.008] [error] Hyprland IPC: Unable to connect?
```

I noticed in the [Zen Browser Flatpak configuration file](https://github.com/zen-browser/desktop/blob/main/flatpak/io.github.zen_browser.zen.yml.template), and I suspect that the AUR version does not have specific permissions and arguments related to Wayland and IPC.

**Details:**

- **Zen Browser version**: [`zen-browser-avx2-bin 1.0.1.a.7-1` package from the Arch AUR](https://aur.archlinux.org/packages/zen-browser-avx2-bin)
- **Waybar version**: `waybar 0.11.0-3` package installed from Arch official repositories.
- **Hyprland version**: `hyprland 0.43.0-1` installed from Arch official repositories

### What I have tested:
- I have verified that the `.socket.sock` and `.socket2.sock` files exist in `/run/user/1000/hypr`.
- The issue persists only when Zen Browser is running. The IPC connection error is resolved if I close Zen Browser and restart Waybar.
- I am using the AUR version of Zen Browser, not the Flatpak version. I noticed in the [Zen Browser Flatpak configuration file](https://github.com/zen-browser/desktop/blob/main/flatpak/io.github.zen_browser.zen.yml.template) that there are specific permissions and arguments related to Wayland and IPC, which are not present in the AUR version.
  
Could the absence of these permissions or configurations in the AUR package be causing the IPC conflict with Waybar?

It would be great if the Zen Browser were available as a package in the official Arch repo. Else, I can relay the information I gather from you to the AUR package maintainer.

**EDIT:** Updated log.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.7 (64-bit)

### What platform are you seeing the problem on?

Linux

### Relevant log output

```shell
[user_redacted][~] → grep -i -E 'ipc|hyprlanda|waybar' /run/user/1000/
hypr/[$HYPRLAND_INSTANCE_SIGNATURE]/hyprland.log
[LOG] Executing bash ~/.config/waybar/scripts/hyprland_event_listener.sh
[LOG] LayerSurface 5fc32bf74f60 (namespace waybar layer 2) created on monitor HDMI-A-1
[LOG] Window 5fc32c71e970 set title to IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser
[LOG] Searching for matching rules for zen-alpha (title: IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser)
[LOG] Window 5fc32c71e970 set title to IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser
[LOG] Searching for matching rules for zen-alpha (title: IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser)
[LOG] Window 5fc32c71e970 set title to IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser
[LOG] Searching for matching rules for zen-alpha (title: IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser)
[LOG] Window 5fc32c71e970 set title to Waybar IPC Connection Error — Zen Browser
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Window 5fc32c71e970 set title to Waybar IPC Connection Error — Zen Browser
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Set keyboard focus to surface 5fc32c6bac40, with [Window 5fc32c71e970: title: "Waybar IPC Connection Error — Zen Browser"]
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Set keyboard focus to surface 5fc32c6bac40, with [Window 5fc32c71e970: title: "Waybar IPC Connection Error — Zen Browser"]
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Set keyboard focus to surface 5fc32c6bac40, with [Window 5fc32c71e970: title: "Waybar IPC Connection Error — Zen Browser"]
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Set keyboard focus to surface 5fc32c6bac40, with [Window 5fc32c71e970: title: "Waybar IPC Connection Error — Zen Browser"]
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Window 5fc32c71e970 set title to IPC – Hyprland Wiki — Zen Browser
[LOG] Searching for matching rules for zen-alpha (title: IPC – Hyprland Wiki — Zen Browser)
[LOG] Window 5fc32c71e970 set title to Waybar IPC Connection Error — Zen Browser
[LOG] Searching for matching rules for zen-alpha (title: Waybar IPC Connection Error — Zen Browser)
[LOG] Window 5fc32c71e970 set title to IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser
[LOG] Searching for matching rules for zen-alpha (title: IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser)
[LOG] Searching for matching rules for zen-alpha (title: IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser)
[LOG] Set keyboard focus to surface 5fc32c6bac40, with [Window 5fc32c71e970: title: "IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser"]
[LOG] Searching for matching rules for zen-alpha (title: IPC Connection Issue with Waybar and Zen Browser (AUR Version) · Issue #1992 · zen-browser/desktop — Zen Browser)
```

**Comment by RayZ3R0** [comment]
Does this still happen in latest version?

**Comment by 0x4a61636f62** [comment]
It seems to have been fixed with `zen-browser-avx2-bin` version `1.0.1.a.12-1`.

**Comment by RayZ3R0** [comment]
Alright, will close this then, thanks for reporting! Also, current version is a.17
